lib/src/conflicts.rs

```diff
@@ -71,36 +110,39 @@ pub fn describe_conflict(conflict: &Conflict, file: &mut dyn Write) -> std::io::
     Ok(())
 }
 
-fn file_terms(terms: &[ConflictTerm]) -> Vec<&ConflictTerm> {
+fn file_terms(terms: &[Option<TreeValue>]) -> Vec<Option<&TreeValue>> {
     terms
         .iter()
-        .filter(|term| {
-            matches!(
-                term.value,
-                TreeValue::File {
-                    executable: false,
-                    ..
-                }
-            )
+        .filter_map(|term| match term {
+            Some(
+                value @ TreeValue::File {
+                    executable: false, ..
+                },
+            ) => Some(Some(value)),
+            None => Some(None),
+            _ => None,
```

Comment:
I think this logic is confusing (it doesn't look correct if one of the files is a symlink, but it works out OK due to how the function is used).

I'm not sure if any action is needed in this PR. A better solution might be to rewrite `extract_file_conflict_as_single_hunk` to make it understand executable bits, in a later PR.

Reply:
I agree that it's confusing, but I think it was the closest I keep it to the old behavior. I'll leave it for later to clean it up along with executable bits as you said.